modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -129,6 +130,245 @@ func (suite *KeeperTestSuite) TestChanUpgradeInit() {
+			},
+			false,
+		},
+		{
+			"channel state is not in OPEN or INITUPGRADE state",
+			func() {
+				suite.Require().NoError(path.EndpointB.SetChannelState(types.CLOSED))
+			},
+			false,
+		},
+		{
+			"timeout has passed",
+			func() {
+				counterpartyUpgrade.Timeout = types.NewTimeout(clienttypes.NewHeight(0, 1), 0)
+			},
+			false,
+		},
+		{
+			"invalid connection state",
+			func() {
+				connectionEnd := path.EndpointB.GetConnection()
+				connectionEnd.State = connectiontypes.UNINITIALIZED
+				suite.chainB.GetSimApp().GetIBCKeeper().ConnectionKeeper.SetConnection(suite.chainB.GetContext(), proposedConnectionHops[0], connectionEnd)
+			},
+			false,
+		},
+		{
+			"invalid connection hops",
+			func() {
+				channel := path.EndpointB.GetChannel()
+				channel.ConnectionHops = []string{"connection-100"}
+				path.EndpointB.SetChannel(channel)
+			},
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			suite.coordinator.Setup(path)
+
+			path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Version = fmt.Sprintf("%s-v2", mock.Version)
+
+			err := path.EndpointA.ChanUpgradeInit()
+			suite.Require().NoError(err)
+
+			// commit a block to update chain A for correct proof querying
+			path.EndpointA.Chain.Coordinator.CommitBlock(path.EndpointA.Chain)
+			// update chainB's client of chain A to account for ChanUpgradeInit
+			suite.Require().NoError(path.EndpointB.UpdateClient())
+
+			proposedConnectionHops = []string{path.EndpointB.ConnectionID}
+			upgrade := types.NewUpgrade(
+				types.NewUpgradeFields(
+					types.UNORDERED, proposedConnectionHops, fmt.Sprintf("%s-v2", mock.Version),
+				),
+				types.NewTimeout(path.EndpointA.Chain.GetTimeoutHeight(), 0),
+				0,
+			)
+			counterpartyUpgrade = path.EndpointA.GetProposedUpgrade()
+			// expSequence = 1
+
+			tc.malleate()
+
+			// we need to update the clients again because malleation has changed the channel state
+			suite.Require().NoError(path.EndpointA.UpdateClient())
+			suite.Require().NoError(path.EndpointB.UpdateClient())
+
+			proofCounterpartyChannel, proofCounterpartyUpgrade, proofHeight := path.EndpointB.QueryChannelUpgradeProof()
+
+			_, err = suite.chainB.GetSimApp().IBCKeeper.ChannelKeeper.ChanUpgradeTry(
+				suite.chainB.GetContext(),
+				path.EndpointB.ChannelConfig.PortID,
+				path.EndpointB.ChannelID,
+				proposedConnectionHops,
+				upgrade.Timeout,
+				counterpartyUpgrade,
+				path.EndpointA.GetChannel().UpgradeSequence,
+				proofCounterpartyChannel,
+				proofCounterpartyUpgrade,
+				proofHeight)
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+				// suite.Require().Equal(expSequence, path.EndpointB.GetChannel().UpgradeSequence)
+				// suite.Require().Equal(mock.Version, path.EndpointB.GetChannel().Version)
+				// suite.Require().Equal(path.EndpointB.GetChannel().State, types.TRYUPGRADE)
```

Comment:
these can be removed, channel version won't change until the upgrade completes and the channel state change will be checked at the msg server testing level